### PR TITLE
support change thin/thick, data_reduction in migration

### PR DESCRIPTION
**Pull request by DingDino**
Support changing thin/thick, data_reduction in volume migration for cinder retype user story

### CODE REVIEW

File: storops/unity/resource/lun.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def _is_move_session_supported(self, dest):
             return False
         return True
 
-    def migrate(self, dest, **kwargs):
+    def migrate(self, dest, provision=None, **kwargs):
```
Comment (Murray-LIANG) [inline storops/unity/resource/lun.py]: Change the name of `provision` to `dest_provision`. And leave a comment about the possible value ('thin', 'thick', 'compressed') of it.

Comment (yong-huang) [inline storops/unity/resource/lun.py]: Why not get provision from kwargs, not suggest to modify the interface if no need.

File: storops/unity/resource/lun.py [hunk 2]
```diff
@@ -388,11 +388,17 @@ def _do_check_move_session(move_session_id):
                 return False
 
         clz = storops.unity.resource.move_session.UnityMoveSession
-        is_compressed = self.is_data_reduction_enabled
-
+        is_compressed = True if provision == 'compressed' else \
+            self.is_data_reduction_enabled
+        is_thin = True
```
Comment (Murray-LIANG) [inline storops/unity/resource/lun.py]: Could use `is_thin = None` when `provision = None` to keep default behavior unchange.

File: storops/unity/resource/lun.py [hunk 3]
```diff
@@ -388,11 +388,17 @@ def _do_check_move_session(move_session_id):
                 return False
 
         clz = storops.unity.resource.move_session.UnityMoveSession
-        is_compressed = self.is_data_reduction_enabled
-
+        is_compressed = True if provision == 'compressed' else \
```
Comment (yong-huang) [inline storops/unity/resource/lun.py]: Move these codes out of storops, get the values of is_thin and is_compressed in cinder driver client.py, and pass them when invoking migrate method.

File: storops/unity/resource/lun.py (hunk 1 above)
Comment (DingDino) [reply storops/unity/resource/lun.py]: will add (None, 'thin', 'thick', 'compressed') possible values in comment.
Move this part to cinder code as suggested.

Comment (DingDino) [reply storops/unity/resource/lun.py]: will fix this

File: storops/unity/resource/lun.py (hunk 3 above)
Comment (DingDino) [reply storops/unity/resource/lun.py]: Move this part to cinder code as suggested.

**Pull request merged**